addons/l10n_in/tests/test_invoice_hsn_summary.py

```diff
@@ -0,0 +1,596 @@
+        ]
+
+        self.assert_hsn_summary(invoice, {
+            'has_igst': True,
+            'has_gst': False,
+            'has_cess': False,
+            'nb_columns': 6,
+            'display_uom': False,
+            'items': [
+                {
+                    'l10n_in_hsn_code': self.test_hsn_code_1,
+                    'quantity': 3.0,
+                    'uom': self.uom_unit,
+                    'rate': 0.0,
+                    'amount_untaxed': 200.0,
+                    'tax_amount_igst': 36.0,
+                    'tax_amount_cgst': 0.0,
+                    'tax_amount_sgst': 0.0,
+                    'tax_amount_cess': 0.0,
+                },
+                {
+                    'l10n_in_hsn_code': self.test_hsn_code_2,
+                    'quantity': 3.0,
+                    'uom': self.uom_unit,
+                    'rate': 0.0,
+                    'amount_untaxed': 200.0,
+                    'tax_amount_igst': 36.0,
+                    'tax_amount_cgst': 0.0,
+                    'tax_amount_sgst': 0.0,
+                    'tax_amount_cess': 0.0,
+                },
+            ],
+        })
+
+    def test_l10n_in_hsn_summary_4(self):
+        """ Zero rated GST or no taxes at all."""
+        invoice = self.create_invoice(invoice_line_ids=[
+            Command.create({
+                'product_id': self.product_a.id,
+                'quantity': 1.0,
+                'price_unit': 350.0,
+                'product_uom_id': self.uom_unit.id,
+                'tax_ids': [],
+            }),
+            Command.create({
+                'product_id': self.product_a.id,
+                'quantity': 1.0,
+                'price_unit': 350.0,
+                'product_uom_id': self.uom_unit.id,
+                'tax_ids': [],
+            }),
+        ])
+
+        self.assert_hsn_summary(invoice, {
+            'has_igst': False,
+            'has_gst': False,
+            'has_cess': False,
+            'nb_columns': 5,
+            'display_uom': False,
+            'items': [
+                {
+                    'l10n_in_hsn_code': self.test_hsn_code_1,
+                    'quantity': 2.0,
+                    'uom': self.uom_unit,
+                    'rate': 0.0,
+                    'amount_untaxed': 700.0,
+                    'tax_amount_igst': 0.0,
+                    'tax_amount_cgst': 0.0,
+                    'tax_amount_sgst': 0.0,
+                    'tax_amount_cess': 0.0,
+                },
+            ],
+        })
+
+        # No tax to IGST 0%/exempt.
+        invoice.invoice_line_ids = [
+            Command.update(invoice.invoice_line_ids[0].id, {'tax_ids': [Command.set(self.igst_0.ids)]}),
+            Command.update(invoice.invoice_line_ids[1].id, {'tax_ids': [Command.set(self.exempt_0.ids)]}),
+        ]
+
+        self.assert_hsn_summary(invoice, {
+            'has_igst': False,
```

Comment:
As mentioned, the flag is `False` because the IGST 0% doesn't have the IGST tag. Idk if it's a bug or not on the indian coa.